tesseract_command_language/include/tesseract_command_language/joint_waypoint.h

```diff
@@ -289,7 +289,7 @@ class JointWaypoint
 
     // Check if they are close to 0
     Eigen::VectorXd range = upper_tolerance - lower_tolerance;
-    double sum = range.sum();
+    double sum = range.cwiseAbs().sum();
     return (sum > 1e-5);
```

Comment:
Why don't you use `Eigen::isApprox(upper_tolerance.cwiseAbs(), lower_tolerance.cwiseAbs())`? The `1e-5` seems arbitrary; if you care about them being the same you should use machine `epsilon` which is think is the default for `Eigen::isApprox`

Reply:
> Eigen::isApprox(upper_tolerance.cwiseAbs(), lower_tolerance.cwiseAbs())

We probably don't want to use cwiseAbs() because -pi would show equal to pi, but I see where you are going.

Suggest changing to: `Eigen::isApprox(upper_tolerance, lower_tolerance)`?

Reply:
I was thinking that `lower_tolerance` might be negative and need to have the absolute value taken. If both vectors are positive then that would work

Reply:
I just assume that `Eigen::isApprox` handles this when checking against the epsilon but not sure if that is true.

Reply:
It looks like the isApprox function should handle this; do you agree?

The fuzzy compares are done multiplicatively. Two vectors v and w are considered to be approximately equal within precision p if

∥v−w∥⩽pmin(∥v∥,∥w∥).
For matrices, the comparison is done using the Hilbert-Schmidt norm (aka Frobenius norm L2 norm).
Because of the multiplicativeness of this comparison, one can't use this function to check whether *this is approximately equal to the zero matrix or vector. Indeed, isApprox(zero) returns false unless *this itself is exactly the zero matrix or vector. If you want to test whether *this is zero, use internal::isMuchSmallerThan(const RealScalar&, RealScalar) instead.

Reply:
Now that I read the description I am not sure we should use this check by itself because of the zero case.

Reply:
It appears that `isApprox` doesn't do element-wise comparison, and it also doesn't make sure both vectors are positive. [Here is the description](https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#ae8443357b808cd393be1b51974213f9c)

Reply:
I don't think this will work if one of the vectors is negative because of the subtraction on the left hand side. If you're going to use this, the vectors both need to be positive. Also if they can be zero vectors (which I assume they are by default), then the original change might be better (with the addition of a comparison to machine epsilon instead of 1e-5)

Reply:
This approach `Eigen::isApprox(upper_tolerance.cwiseAbs(), lower_tolerance.cwiseAbs())` fails in the unit test. The following `!Eigen::IsApprox(lower_tolerance, upper_tolerance)` works but based on the description it does pose concern for the all zero case. I created a vectorized version of `almostEqualRelativeAndAbs` which is probably the best solution.

``` c++
  std::vector<std::string> joint_names = {"joint_1", "joint_2", "joint_3"};
  Eigen::VectorXd joint_values = Eigen::VectorXd::Constant(3, 1);

  JointWaypoint jw(joint_names, joint_values);
  EXPECT_FALSE(jw.isToleranced());

  jw.upper_tolerance = Eigen::VectorXd::Constant(3, 5);
  jw.lower_tolerance = Eigen::VectorXd::Constant(3, -5);
  EXPECT_TRUE(jw.isToleranced());

  jw.upper_tolerance = Eigen::VectorXd::Constant(3, 0);
  jw.lower_tolerance = Eigen::VectorXd::Constant(3, 0);
  EXPECT_FALSE(jw.isToleranced());
```

Reply:
I update the PR to leverage the `almostEqualRelativeAndAbs` function.